src/bin/cargo/cli.rs

```diff
@@ -107,26 +107,51 @@ fn expand_aliases(
             commands::builtin_exec(cmd),
             super::aliased_command(config, cmd)?,
         ) {
-            (None, Some(mut alias)) => {
-                alias.extend(
+            (None, Some(mut user_alias)) => {
+                user_alias.extend(
                     args.values_of("")
                         .unwrap_or_default()
                         .map(|s| s.to_string()),
                 );
                 let args = cli()
                     .setting(AppSettings::NoBinaryName)
-                    .get_matches_from_safe(alias)?;
+                    .get_matches_from_safe(user_alias)?;
+                return expand_aliases(config, args);
+            }
+            (
+                Some(BuiltinExec {
+                    alias_for: Some(alias_for),
+                    ..
+                }),
+                Some(mut user_alias),
+            ) => {
+                // User alias conflicts with built-in alias
+                config.shell().warn(format!(
+                    "user-defined alias `{}` overrides a built-in alias for `{}`",
```

Comment:
I don't think a warning is always necessary here, since it would warn every time you run the command, it could get quite noisy.  Maybe only issue the warning if `config.extra_verbose()` is set?  Or just remove it?

Reply:
Sorry for the delay.

I just removed this warning.
Because this PR stops using Clap alias, options including `-vv` are not handled when built-in aliases are used.